python/plugins/processing/algs/gdal/AssignProjection.py

```diff
@@ -110,7 +110,7 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         self.setOutputValue(self.OUTPUT, fileName)
 
         return [
-            self.commandName() + (".bat" if isWindows() else ".py"),
+            self.commandName(),
```

Comment:
@nicogodet 's comment about supporting older GDAL versions might be relevant. What about creating a helper function that use the old or new behaviour depending on the GDAL version ?
Cf https://github.com/qgis/QGIS/commit/06291f790c6a6a366d6ac97ab8a73a0c49d85b7a  for a way of checking the GDAL version from python

Reply:
I deleted it as only Windows might be concerned by this changes and osgeo4w ships 3.9...

I can repost if needed

Reply:
> I deleted it as only Windows might be concerned

Unix might be too for GDAL < 3.9 since launcher scripts will not be present

Reply:
> Can the minimum GDAL version be bumped?

probably a bit, but I doubt we can afford 3.9 yet...

Reply:
OK let's revisit this when 3.9 is the minimum. I don't feel comfortable making larger changes to QGIS to support this. Would be good to note somewhere that the processing features only work with OSGeo4W on Windows....